src/qt/sendcoinsdialog.cpp

```diff
@@ -243,6 +245,72 @@ void SendCoinsDialog::on_sendButton_clicked()
     }
 
     fNewRecipientAllowed = false;
+
+    if (have_warning) {
+        struct prior_usage_info_t {
+            CAmount total_amount{0};
+            int num_txs{0};
+            qint64 tx_time_oldest;
+            qint64 tx_time_newest;
+        };
+        QMap<QString, prior_usage_info_t> prior_usage_info;
+        {
+            QStringList addresses;
+            for (const auto& recipient : recipients) {
+#ifdef ENABLE_BIP70
+                if (recipient.paymentRequest.IsInitialized()) continue;
+#endif
+                addresses.append(recipient.address);
+            }
+            model->findAddressUsage(addresses, [&prior_usage_info](const QString& address, const interfaces::WalletTx& wtx, uint32_t output_index){
+                auto& info = prior_usage_info[address];
+                info.total_amount += wtx.tx->vout[output_index].nValue;
+                ++info.num_txs;
+                if (info.num_txs == 1 || wtx.time < info.tx_time_oldest) {
+                    info.tx_time_oldest = wtx.time;
+                }
+                if (info.num_txs == 1 || wtx.time > info.tx_time_newest) {
+                    info.tx_time_newest = wtx.time;
+                }
+            });
+        }
+
+        QString reuse_question;
+        if (recipients.size() > 1) {
+            reuse_question = tr("You've already paid some of these addresses. Are you sure you want to send again?");
+        } else {
+            reuse_question = tr("You've already paid this address. Are you sure you want to send again?");
+        }
+        reuse_question.append("<br /><span style='font-size:10pt;'>");
+        reuse_question.append(tr("Bitcoin addresses are intended to only be used once, for a single payment."));
+        reuse_question.append("</span><br />");
+
+        for (const auto& rcp : recipients) {
+#ifdef ENABLE_BIP70
+            if (rcp.paymentRequest.IsInitialized()) continue;
+#endif
+            if (!prior_usage_info.contains(rcp.address)) continue;
+            const auto& rcp_prior_usage_info = prior_usage_info.value(rcp.address);
+            const QString label_and_address = rcp.label.isEmpty() ? rcp.address : (rcp.label + " (" + rcp.address + ")");
+            reuse_question.append("<br />");
+            if (rcp_prior_usage_info.num_txs == 1) {
+                //: %1 is an amount (eg, "1 BTC"); %2 is a Bitcoin address and its label; %3 is a date (eg, "2019-05-08")
```

Comment:
Note: `//:` is how Qt lets us add notes for translators. (I'm not sure if it survives to Transifex?)

Reply:
> Note: `//:` is how Qt lets us add notes for translators. (I'm not sure if it survives to Transifex?)

Smth like #21465 is required for that.